docs/DeveloperGuide.md

```diff
@@ -2,28 +2,131 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
+original source as well}
 
 ## Design & implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Help menu feature
+
+#### Implementation
+
+The "help" feature is facilitated by the  `Help` class.
+It provides a static method `printHelp` to print out a guide on how to use the commands in the application.
+`printHelp` can be used in the event the user issues an invalid command
+
+### Group Creation feature
+
+#### Implementation
+
+The "Group Creation" feature is facilitated by the `Group` class. It provides methods to create a new group and manage
+group membership. The implementation of this feature is as follows:
+
+The Group class maintains a list of members as a `private List<User>` field called `members`.
+
+The `createGroup(String groupName)` method is responsible for creating a new group. It performs the following steps:
+
+1. Checks if a group with the given `groupName` already exists using the `isGroup(String groupName)` method.
+2. If the group does not exist, creates a new `Group` object with the provided `groupName`.
+3. Prints a success message indicating that the group has been created.
+4. Adds the new `Group` object to the `groups` list.
+5. Returns the newly created `Group` object.
+6. If the group already exists, prints an error message indicating that the group already exists.
+
+### Add Member to Group feature
+
+#### Implementation
+
+The "Add Member to Group" feature is facilitated by the `Group` class. It provides methods to manage group membership
+and allows users to add new members to an existing group. The implementation of this feature is as follows:
+
+The Group class maintains a list of members as a `private List<User>` field called `members`.
+
+Users can add a new member to the group by using the command `member USER_NAME`. The `addMember(String memberName)` method is responsible for adding a new member to the group. It performs the following steps:
+
+1. Checks if a user with the given `memberName` is already a member of the group using the `isMember(String memberName)`
+   method.
+2. If the user is not a member, creates a new `User` object with the provided `memberName`.
+3. Adds the new `User` object to the `members` list.
+4. Prints a success message indicating that the member has been added to the group.
+
+![Sequence Diagram](AddMember.png)
+
```

Comment:
Activation bar of Group should not extend beyond the [else] part of alt frame, as Group is not being used.